### PR TITLE
Added new tests

### PR DESCRIPTION
**Pull request by Pavellys**

### CODE REVIEW

File: src/main/java/constants/FeedConstants.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package constants;
```
Comment (vchesnakoutest) [inline src/main/java/constants/FeedConstants.java]: rename package to "locators_constants"

File: src/main/java/constants/FeedConstants.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+
+public interface FeedConstants {
+    String UPDATE_MOOD_BUTTON_CSS = "#LinkUpdate";
+    String GET_MOOD_LEVEL_RESULT = "//h5[@class='media-heading']//span";
```
Comment (vchesnakoutest) [inline src/main/java/constants/FeedConstants.java]: //span[contains(@class,'badge')]

File: src/main/java/constants/RateYourHappinessConstants.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package constants;
+
+public interface RateYourHappinessConstants {
+    String SLIDER_CSS = ".ui-slider-handle";
+    String TEXTAREA_CSS = "#TextBoxUpdateMoodTag";
+    String RATE_UPDATE_MOOD_BUTTON = "//*[@class='btn btn-sm btn-primary ButtonUpdate']";
```
Comment (vchesnakoutest) [inline src/main/java/constants/RateYourHappinessConstants.java]: //*[contains(@class,'ButtonUpdate')]

File: src/main/java/constants/RateYourHappinessConstants.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+    String BUTTON_MY_DIARY = "//*[@class='btn btn-sm btn-default ButtonMyDiary pull-left']";
+    String HUGS_BUTTON = "//span[@class='ButtonHug']";
```
Comment (vchesnakoutest) [inline src/main/java/constants/RateYourHappinessConstants.java]: //*[@class='ButtonHug']

File: src/main/java/pages/FeedPage.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+import com.codeborne.selenide.Condition;
+import constants.FeedConstants;
+import java.time.Duration;
+import static com.codeborne.selenide.Selectors.byXpath;
+import static com.codeborne.selenide.Selenide.$;
+
+public class FeedPage implements FeedConstants {
+
+        public pages.FeedPage waitForFeedPageOpened(){
```
Comment (vchesnakoutest) [inline src/main/java/pages/FeedPage.java]: just public FeedPage

File: src/main/java/pages/LoginPage.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+    public LoginPage waitForLoginFormOpened(){
+        $(GOOGLE_PLAY_ICON_CSS).shouldBe(Condition.visible, Duration.ofSeconds(20));
+        return this;
+    }
+
+    public String getCurrentURL(){
+        return url();
+    }
+
```
Comment (vchesnakoutest) [inline src/main/java/pages/LoginPage.java]: remove empty line

File: src/main/java/pages/RateYourHappiness.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
+package pages;
+
+import com.codeborne.selenide.Condition;
+import constants.RateYourHappinessConstants;
+import org.openqa.selenium.Keys;
+import static com.codeborne.selenide.Selectors.byXpath;
+import static com.codeborne.selenide.Selenide.$;
+
+public class RateYourHappiness implements RateYourHappinessConstants {
```
Comment (vchesnakoutest) [inline src/main/java/pages/RateYourHappiness.java]: divide on several modal pages

File: src/main/java/pages/RateYourHappiness.java [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+        }else{
+            direction = Keys.ARROW_LEFT;
+        }
+        for(int i = 0; i < Math.abs(moodValue - defaultMood) ; i++){
+           $(SLIDER_CSS).sendKeys(direction);
+        }
+        return this;
+    }
+
+    public RateYourHappiness description(String text){
```
Comment (vchesnakoutest) [inline src/main/java/pages/RateYourHappiness.java]: fillInDescriptionInput

File: src/main/java/pages/RateYourHappiness.java [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+    public RateYourHappiness clickToHug(){
+        $(byXpath(HUGS_BUTTON)).click();
+        return this;
+    }
+
+    public RateYourHappiness waitForHug(){
+        $(byXpath(HUGS_QUANTITY)).shouldHave(Condition.exactText("1"));
+        return this;
+    }
```
Comment (vchesnakoutest) [inline src/main/java/pages/RateYourHappiness.java]: move to FeedPage

File: src/main/java/pages/RateYourHappiness.java [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+
+    public String getHugQuantity(){
+        return $(byXpath(HUGS_QUANTITY)).getText();
+
```
Comment (vchesnakoutest) [inline src/main/java/pages/RateYourHappiness.java]: remove empty line

**Review by vchesnakoutest — CHANGES_REQUESTED** [review]
please make fixes

File: src/main/java/constants/FeedConstants.java (hunk 1 above)
Comment (Pavellys) [reply src/main/java/constants/FeedConstants.java]: done

File: src/main/java/constants/FeedConstants.java (hunk 2 above)
Comment (Pavellys) [reply src/main/java/constants/FeedConstants.java]: done

File: src/main/java/constants/RateYourHappinessConstants.java (hunk 3 above)
Comment (Pavellys) [reply src/main/java/constants/RateYourHappinessConstants.java]: done

File: src/main/java/constants/RateYourHappinessConstants.java (hunk 4 above)
Comment (Pavellys) [reply src/main/java/constants/RateYourHappinessConstants.java]: done

File: src/main/java/pages/FeedPage.java (hunk 5 above)
Comment (Pavellys) [reply src/main/java/pages/FeedPage.java]: done

File: src/main/java/pages/LoginPage.java (hunk 6 above)
Comment (Pavellys) [reply src/main/java/pages/LoginPage.java]: done

File: src/main/java/pages/RateYourHappiness.java (hunk 8 above)
Comment (Pavellys) [reply src/main/java/pages/RateYourHappiness.java]: done

File: src/main/java/pages/RateYourHappiness.java (hunk 10 above)
Comment (Pavellys) [reply src/main/java/pages/RateYourHappiness.java]: done

File: src/main/java/pages/RateYourHappiness.java (hunk 9 above)
Comment (Pavellys) [reply src/main/java/pages/RateYourHappiness.java]: Done

File: src/main/java/pages/RateYourHappiness.java (hunk 7 above)
Comment (Pavellys) [reply src/main/java/pages/RateYourHappiness.java]: done

File: MoodPanda.iml [hunk 11]
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (vchesnakoutest) [inline MoodPanda.iml]: don't commit iml files to project

File: src/main/java/locators_constants/MoodUpdatedModalConstants.java [hunk 12]
```diff
@@ -0,0 +1,5 @@
+package locators_constants;
+
+public interface MoodUpdatedModalConstants {
+    String BUTTON_MY_DIARY = "//*[@class='btn btn-sm btn-default ButtonMyDiary pull-left']";
```
Comment (vchesnakoutest) [inline src/main/java/locators_constants/MoodUpdatedModalConstants.java]: don't include all classes in xpath, we discussed that on previous lessons,  better to use something like that:
//*[contains(@class, 'ButtonMyDiary')]

File: src/main/java/pages/FeedPage.java [hunk 13]
```diff
@@ -1,25 +1,39 @@
...
             return $(byXpath(GET_MOOD_LEVEL_RESULT)).getText();
         }
 
+    public FeedPage clickToHug(){
+        $(byXpath(HUGS_BUTTON)).click();
+        return this;
+    }
+
+    public FeedPage waitForHug(){
+        $(byXpath(HUGS_QUANTITY)).shouldHave(Condition.exactText("1"));
```
Comment (vchesnakoutest) [inline src/main/java/pages/FeedPage.java]: don't hardcode test values in methods (I mean value "1"). If you want to wait for the specific value, pass this as parameter in method and test.

waitForHugsQuantityDisplayed(String hugsQuantity)

File: src/test/java/locators_constants/Constants.java [hunk 14]
```diff
@@ -1,4 +1,4 @@
-package constants;
+package locators_constants;
```
Comment (vchesnakoutest) [inline src/test/java/locators_constants/Constants.java]: this class contains Test data, so it's not "locators_constants"

File: src/main/java/pages/RateYourHappinessModal.java [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+        }else{
+            direction = Keys.ARROW_LEFT;
+        }
+        for(int i = 0; i < Math.abs(moodValue - defaultMood) ; i++){
+           $(SLIDER_CSS).sendKeys(direction);
+        }
+        return this;
+    }
+
+    public RateYourHappinessModal description(String text){
```
Comment (vchesnakoutest) [inline src/main/java/pages/RateYourHappinessModal.java]: fillInDescriptionInput

**Review by vchesnakoutest — CHANGES_REQUESTED** [review]
please make fixes

File: MoodPanda.iml (hunk 11 above)
Comment (Pavellys) [reply MoodPanda.iml]: done

File: src/main/java/locators_constants/MoodUpdatedModalConstants.java (hunk 12 above)
Comment (Pavellys) [reply src/main/java/locators_constants/MoodUpdatedModalConstants.java]: done

File: src/main/java/pages/FeedPage.java (hunk 13 above)
Comment (Pavellys) [reply src/main/java/pages/FeedPage.java]: done

File: src/test/java/locators_constants/Constants.java (hunk 14 above)
Comment (Pavellys) [reply src/test/java/locators_constants/Constants.java]: done

File: src/main/java/pages/RateYourHappinessModal.java (hunk 15 above)
Comment (Pavellys) [reply src/main/java/pages/RateYourHappinessModal.java]: i deleted that method, it's  not need

Comment (Pavellys) [reply src/main/java/pages/RateYourHappinessModal.java]: done

**Review by vchesnakoutest — APPROVED** [review]

**Pull request merged**